src/mongocxx/test/spec/util.cpp

```diff
@@ -665,134 +835,161 @@ void run_transaction_operations(document::view test,
+                    // (if applicable).
+                    apm_checker.clear();
+                    apm_checker.set_ignore_command_monitoring_event("configureFailPoint");
+
+                    // If a test uses `targetedFailPoint`, disable the fail point after running all
+                    // `operations` to avoid spurious failures in subsequent tests.
+                    stdx::optional<targeted_fail_point_guard_type> targeted_fail_point_guard;
+
+                    // Step 11. Perform the operations.
+                    run_transaction_operations(test.get_document().value,
+                                               &client,
+                                               db_name,
+                                               coll_name,
+                                               &session0,
+                                               &session1,
+                                               &targeted_fail_point_guard,
+                                               apm_checker);
+
+                    // Step 12: Call session0.endSession() and session1.endSession (via
+                    // client_session dtors).
+                }
+
+                // Step 13. If the test includes a list of command-started events in expectations,
+                // compare them to the actual command-started events using the same logic as the
+                // legacy Command Monitoring Spec Tests runner, plus the rules in the
+                // Command-Started Events instructions.
+                test_util::match_visitor visitor =
+                    [&](bsoncxx::stdx::string_view key,
+                        bsoncxx::stdx::optional<bsoncxx::types::bson_value::view> main,
+                        bsoncxx::types::bson_value::view pattern) {
+                        if (key.compare("lsid") == 0) {
+                            REQUIRE(pattern.type() == type::k_string);
+                            REQUIRE(main);
+                            REQUIRE(main->type() == type::k_document);
+                            auto session_name = pattern.get_string().value;
+                            if (session_name.compare("session0") == 0) {
+                                REQUIRE(
+                                    test_util::matches(session_lsid0, main->get_document().value));
+                            } else {
+                                REQUIRE(
+                                    test_util::matches(session_lsid1, main->get_document().value));
+                            }
+                        } else if (pattern.type() == type::k_null) {
+                            if (main) {
+                                return test_util::match_action::k_not_equal;
+                            }
+                            return test_util::match_action::k_skip;
+                        } else if (key.compare("upsert") == 0 || key.compare("multi")) {
```

Comment:
```suggestion
                        } else if (key.compare("upsert") == 0 || key.compare("multi") == 0) {
```

I goofed in the [previous suggestion](https://github.com/mongodb/mongo-cxx-driver/pull/1007#discussion_r1288986161). Fixing this results in other test failures in matching.

The test failures may suggest missing options or operations. [This change](https://github.com/mongodb/mongo-cxx-driver/commit/b1c1ba33b2f95ff9ebf4c3de9b4e300139879062) fixes the "causal consistency disabled" test.